Fix Flow Dependencies computation for flows with dynamic defaults

### Describe the issue

As of 1.0, due to a bug, any flow with dynamic defaults, such as the one below, will be ignored by the Dependencies view:

```yaml
id: infrastructure_workflow
namespace: anna

inputs:
  - id: release_env
    displayName: Release Environment
    type: SELECT
    autoSelectFirst: true
    values:
      - DEV
      - TEST
      - QA
      - PROD

  - id: backup_enabled
    type: BOOL
    defaults: "{{ inputs.release_env == 'PROD' ? true : false }}"
    dependsOn:
      inputs:
        - release_env

tasks:
  - id: log
    type: io.kestra.plugin.core.log.Log
    message: dynamic default {{ inputs.backup_enabled }}
```

We should find a solution so that Flow dependencies from triggers are computed correctly - example from the original issue https://github.com/kestra-io/kestra/issues/11096:

```yaml
id: flow1
namespace: anna

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: Hello World! 🚀
```

another flow:

```yaml
id: flow2
namespace: anna

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: Hello World! 🚀

triggers:
  - id: release
    type: io.kestra.plugin.core.trigger.Flow
    states:
      - SUCCESS
    preconditions:
      id: flows
      flows:
        - namespace: anna
          flowId: flow1
```

### Environment

- Kestra Version: develop

failing cause: 

```
12:31:24.023 ERROR jdbc-queue-FlowInterface_1 i.k.c.topologies.FlowTopologyService unable to validate condition in FlowTopologyService, flow: infrastructure_workflow, condition: io.kestra.plugin.core.trigger.Flow$UpstreamFlowCondition@d6b163f
java.lang.RuntimeException: Unable to inject default value for input 'backup_enabled'
	at io.kestra.core.runners.RunVariables$DefaultBuilder.lambda$build$4(RunVariables.java:291)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at io.kestra.core.runners.RunVariables$DefaultBuilder.build(RunVariables.java:287)
	at io.kestra.core.runners.RunContextFactory.of(RunContextFactory.java:102)
	at io.kestra.core.runners.RunContextFactory.of(RunContextFactory.java:83)
	at io.kestra.core.services.ConditionService.isValid(ConditionService.java:38)
	at io.kestra.core.services.ConditionService.isValid(ConditionService.java:48)
	at io.kestra.core.topologies.FlowTopologyService.validateCondition(FlowTopologyService.java:213)
	at io.kestra.core.topologies.FlowTopologyService.lambda$validateMultipleConditions$17(FlowTopologyService.java:232)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.tryAdvance(AbstractList.java:708)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at io.kestra.core.topologies.FlowTopologyService.validateMultipleConditions(FlowTopologyService.java:232)
	at io.kestra.core.topologies.FlowTopologyService.lambda$isTriggerChild$14(FlowTopologyService.java:198)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.tryAdvance(AbstractList.java:708)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at io.kestra.core.topologies.FlowTopologyService.isTriggerChild(FlowTopologyService.java:198)
	at io.kestra.core.topologies.FlowTopologyService.isChild(FlowTopologyService.java:152)
	at io.kestra.core.topologies.FlowTopologyService.map(FlowTopologyService.java:130)
	at io.kestra.core.topologies.FlowTopologyService.lambda$topology$5(FlowTopologyService.java:118)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at io.kestra.jdbc.runner.JdbcExecutor.lambda$run$13(JdbcExecutor.java:401)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at io.kestra.jdbc.runner.JdbcQueue.lambda$receive$11(JdbcQueue.java:332)
	at io.kestra.jdbc.runner.JdbcQueue.lambda$receiveImpl$17(JdbcQueue.java:405)
	at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:93)
	at io.kestra.jdbc.runner.JdbcQueue.lambda$receiveImpl$18(JdbcQueue.java:385)
	at io.kestra.jdbc.runner.JdbcQueue.lambda$poll$20(JdbcQueue.java:438)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.kestra.core.exceptions.IllegalVariableEvaluationException: io.pebbletemplates.pebble.error.PebbleException: Could not perform equals comparison ({{ inputs.release_env == 'PROD' ? true : false }}:1)
	at io.kestra.core.runners.VariableRenderer.properPebbleException(VariableRenderer.java:129)
	at io.kestra.core.runners.VariableRenderer.renderOnce(VariableRenderer.java:185)
	at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:156)
	at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:141)
	at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:133)
	at io.kestra.core.models.property.PropertyContext$1.render(PropertyContext.java:29)
	at io.kestra.core.models.property.Property.as(Property.java:161)
	at io.kestra.core.models.property.Property.as(Property.java:149)
	at io.kestra.core.runners.FlowInputOutput.resolveDefaultPropertyAs(FlowInputOutput.java:373)
	at io.kestra.core.runners.FlowInputOutput.resolveDefaultValue(FlowInputOutput.java:359)
	at io.kestra.core.runners.RunVariables$DefaultBuilder.lambda$build$4(RunVariables.java:289)
	... 63 common frames omitted
Caused by: io.pebbletemplates.pebble.error.PebbleException: Could not perform equals comparison ({{ inputs.release_env == 'PROD' ? true : false }}:1)
	at io.pebbletemplates.pebble.node.expression.EqualsExpression.evaluate(EqualsExpression.java:26)
	at io.pebbletemplates.pebble.node.expression.EqualsExpression.evaluate(EqualsExpression.java:16)
	at io.pebbletemplates.pebble.node.expression.TernaryExpression.evaluate(TernaryExpression.java:38)
	at io.pebbletemplates.pebble.node.PrintNode.render(PrintNode.java:37)
	at io.pebbletemplates.pebble.node.BodyNode.render(BodyNode.java:44)
	at io.pebbletemplates.pebble.node.RootNode.render(RootNode.java:31)
	at io.pebbletemplates.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:157)
	at io.pebbletemplates.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:96)
	at io.kestra.core.runners.VariableRenderer.renderOnce(VariableRenderer.java:179)
	... 72 common frames omitted
Caused by: io.pebbletemplates.pebble.error.RootAttributeNotFoundException: Root attribute [inputs] does not exist or can not be accessed and strict variables is set to true. ({{ inputs.release_env == 'PROD' ? true : false }}:1)
	at io.pebbletemplates.pebble.node.expression.ContextVariableExpression.evaluate(ContextVariableExpression.java:44)
	at io.pebbletemplates.pebble.node.expression.GetAttributeExpression.evaluate(GetAttributeExpression.java:62)
	at io.pebbletemplates.pebble.node.expression.EqualsExpression.evaluate(EqualsExpression.java:22)
	... 80 common frames omitted
```

--- a/core/src/main/java/io/kestra/core/runners/RunVariables.java
+++ b/core/src/main/java/io/kestra/core/runners/RunVariables.java
@@ -10,6 +10,7 @@ import io.kestra.core.models.flows.FlowInterface;
 import io.kestra.core.models.flows.Input;
 import io.kestra.core.models.flows.State;
 import io.kestra.core.models.flows.input.SecretInput;
+import io.kestra.core.models.property.Property;
 import io.kestra.core.models.property.PropertyContext;
 import io.kestra.core.models.tasks.Task;
 import io.kestra.core.models.triggers.AbstractTrigger;
@@ -282,15 +283,15 @@ public final class RunVariables {
 
                 if (flow != null && flow.getInputs() != null) {
                     // we add default inputs value from the flow if not already set, this will be useful for triggers
-                        flow.getInputs().stream()
-                            .filter(input -> input.getDefaults() != null && !inputs.containsKey(input.getId()))
-                            .forEach(input -> {
-                                try {
-                                    inputs.put(input.getId(), FlowInputOutput.resolveDefaultValue(input, propertyContext));
-                                } catch (IllegalVariableEvaluationException e) {
-                                    throw new RuntimeException("Unable to inject default value for input '" + input.getId() + "'", e);
-                                }
-                            });
+                    flow.getInputs().stream()
+                        .filter(input -> input.getDefaults() != null && !inputs.containsKey(input.getId()))
+                        .forEach(input -> {
+                            try {
+                                inputs.put(input.getId(), FlowInputOutput.resolveDefaultValue(input, propertyContext));
+                            } catch (IllegalVariableEvaluationException e) {
+                                // Silent catch, if an input depends on another input, or a variable that is populated at runtime / input filling time, we can't resolve it here.
+                            }
+                        });
                 }
 
                 if (!inputs.isEmpty()) {

--- a/core/src/test/java/io/kestra/core/runners/RunVariablesTest.java
+++ b/core/src/test/java/io/kestra/core/runners/RunVariablesTest.java
@@ -1,13 +1,24 @@
 package io.kestra.core.runners;
 
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.flows.DependsOn;
 import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.flows.Type;
+import io.kestra.core.models.flows.input.BoolInput;
+import io.kestra.core.models.property.Property;
 import io.kestra.core.models.property.PropertyContext;
 import io.kestra.core.models.tasks.Task;
 import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.runners.pebble.functions.SecretFunction;
+import io.kestra.core.utils.IdUtils;
+import io.micronaut.context.ApplicationContext;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -111,5 +122,26 @@ class RunVariablesTest {
         assertThat(kestra).hasSize(2);
         assertThat(kestra.get("environment")).isEqualTo("test");
         assertThat(kestra.get("url")).isEqualTo("http://localhost:8080");
+    }
+
+    @Test
+    void nonResolvableDynamicInputsShouldBeSkipped() throws IllegalVariableEvaluationException {
+        Map<String, Object> variables = new RunVariables.DefaultBuilder()
+            .withFlow(Flow
+                .builder()
+                .namespace("a.b")
+                .id("c")
+                .inputs(List.of(
+                    BoolInput.builder().id("a").type(Type.BOOL).defaults(Property.ofValue(true)).build(),
+                    BoolInput.builder().id("b").type(Type.BOOL).dependsOn(new DependsOn(List.of("a"), null)).defaults(Property.ofExpression("{{inputs.a == true}}")).build()
+                ))
+                .build()
+            )
+            .withExecution(Execution.builder().id(IdUtils.create()).build())
+            .build(new RunContextLogger(), PropertyContext.create(new VariableRenderer(Mockito.mock(ApplicationContext.class), Mockito.mock(VariableRenderer.VariableConfiguration.class), Collections.emptyList())));
+
+        Assertions.assertEquals(Map.of(
+            "a", true
+        ), variables.get("inputs"));
     }
 }

--- a/core/src/test/java/io/kestra/core/topologies/FlowTopologyTest.java
+++ b/core/src/test/java/io/kestra/core/topologies/FlowTopologyTest.java
@@ -199,6 +199,75 @@ public class FlowTopologyTest {
 
     }
 
+    @Test
+    void flowTriggerWithTargetFlow() throws FlowProcessingException {
+        // Given
+        var tenantId = randomTenantId();
+        var parent = flowService.importFlow(tenantId,
+            """
+                id: parent
+                namespace: io.kestra.unittest
+                inputs:
+                  - id: a
+                    type: BOOL
+                    defaults: true
+                
+                  - id: b
+                    type: BOOL
+                    defaults: "{{ inputs.a == true }}"
+                    dependsOn:
+                      inputs:
+                        - a
+                tasks:
+                  - id: helloA
+                    type: io.kestra.plugin.core.log.Log
+                    message: Hello A
+                """);
+        var child = flowService.importFlow(tenantId, """
+            id: child
+            namespace: io.kestra.unittest
+            tasks:
+              - id: helloB
+                type: io.kestra.plugin.core.log.Log
+                message: Hello B
+            triggers:
+              - id: release
+                type: io.kestra.plugin.core.trigger.Flow
+                states:
+                  - SUCCESS
+                preconditions:
+                  id: flows
+                  flows:
+                   - namespace: io.kestra.unittest
+                     flowId: parent
+            """);
+        var unrelatedFlow = flowService.importFlow(tenantId, """
+            id: unrelated_flow
+            namespace: io.kestra.unittest
+            tasks:
+              - id: download
+                type: io.kestra.plugin.core.http.Download
+            """);
+
+        // When
+        computeAndSaveTopologies(List.of(child, parent, unrelatedFlow));
+        System.out.println();
+        flowTopologyRepository.findAll(tenantId).forEach(topology -> {
+            System.out.println(FlowTopologyTestData.of(topology));
+        });
+
+        var dependencies = flowService.findDependencies(tenantId, "io.kestra.unittest", parent.getId(), false, true);
+        flowTopologyRepository.findAll(tenantId).forEach(topology -> {
+            System.out.println(FlowTopologyTestData.of(topology));
+        });
+
+        // Then
+        assertThat(dependencies.map(FlowTopologyTestData::of))
+            .containsExactlyInAnyOrder(
+                new FlowTopologyTestData(parent, child)
+            );
+    }
+
     /**
      * this function mimics the production behaviour
      */

